Update libtagarray to 0.0.7

foxtran opened the pull request:
Closes #30 

Most of changes are related to procedures renaming

Commit:
Bump libtagarray version to 0.0.7

Commit:
Update drivers

Commit:
Update modules

karmachoi commented:
I got errors on my Intel Mac. (gcc@12, python3.10)

cheolhochoi@Home:~/clone/openqp$ openqp --run_tests other
[Home.local:38052] shmem: mmap: an error occurred while determining whether or not /var/folders/wp/d5hj4lnj3537wvc_kghx6fpc0000gn/T//ompi.Home.501/jf.0/3917152256/sm_segment.Home.501.e97b0000.0 could be created.

PyOQP: dftd4 is not available
[OQPTester] Starting OpenQP tests for: other
[Home.local:38057] shmem: mmap: an error occurred while determining whether or not /var/folders/wp/d5hj4lnj3537wvc_kghx6fpc0000gn/T//ompi.Home.501/jf.0/1890320384/sm_segment.Home.501.70ac0000.0 could be created.
[Home.local:38055] shmem: mmap: an error occurred while determining whether or not /var/folders/wp/d5hj4lnj3537wvc_kghx6fpc0000gn/T//ompi.Home.501/jf.0/3577675776/sm_segment.Home.501.d53f0000.0 could be created.
[Home.local:38056] shmem: mmap: an error occurred while determining whether or not /var/folders/wp/d5hj4lnj3537wvc_kghx6fpc0000gn/T//ompi.Home.501/jf.0/569704448/sm_segment.Home.501.21f50000.0 could be created.
[Home.local:38058] shmem: mmap: an error occurred while determining whether or not /var/folders/wp/d5hj4lnj3537wvc_kghx6fpc0000gn/T//ompi.Home.501/jf.0/2195324928/sm_segment.Home.501.82da0000.0 could be created.

PyOQP: dftd4 is not available

PyOQP: dftd4 is not available
PyOQP: dftd4 is not available


PyOQP: dftd4 is not available
[OQPTester] Running test for h2o_rohf_mrsf-q_6-31g_dtcam-b3lyp
[Home:38058] *** Process received signal ***
[Home:38058] Signal: Segmentation fault: 11 (11)
[Home:38058] Signal code: Address not mapped (1)
[Home:38058] Failing at address: 0x0
[Home:38058] [ 0] 0   libsystem_platform.dylib            0x00007ff815872e9d _sigtramp + 29
[Home:38058] [ 1] 0   Python                              0x0000000108f07278 list_dealloc + 198
[Home:38058] *** End of error message ***
[OQPTester] Running test for h2o_nacme_rohf_mrsf-s_6-31g_bhhlyp
Traceback (most recent call last):
  File "/usr/local/bin/openqp", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.10/site-packages/oqp/pyoqp.py", line 201, in main
    report, status = run_tests(args.run_tests)
  File "/usr/local/lib/python3.10/site-packages/oqp/pyoqp.py", line 257, in run_tests
    return tester.run(test_path), tester.status
  File "/usr/local/lib/python3.10/site-packages/oqp/utils/oqp_tester.py", line 288, in run
    self.run_tests(test_path)
  File "/usr/local/lib/python3.10/site-packages/oqp/utils/oqp_tester.py", line 186, in run_tests
    self.results.append(future.result())
  File "/usr/local/Cellar/python@3.10/3.10.16/Frameworks/Python.framework/Versions/3.10/lib/python3.10/concurrent/futures/_base.py", line 451, in result
    return self.__get_result()
  File "/usr/local/Cellar/python@3.10/3.10.16/Frameworks/Python.framework/Versions/3.10/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
concurrent.futures.process.BrokenProcessPool: A process in the process pool was terminated abruptly while the future was running or pending.

toxtran commented:
@karmachoi, did you see this error before this patch? It looks like a fail of OpenMPI

karmachoi commented:
> @karmachoi, did you see this error before this patch? It looks like a fail of OpenMPI

I just recompiled this branch. I keep getting this issue.

toxtran commented:
Do you see this error with main brabch?

karmachoi commented:
> Do you see this error with main branch?

No. It only occurs in the updated branch.

karmachoi commented:
I also get error of

/opt/soft/install/binutils/2.37-GCCcore-11.2.0/bin/ld.gold: error: /bighome/cheolho.choi/clone/openqp/build/external/libtagarray/src/libtagarray-build-GNU/libtagarray.a(Utils.cpp.o): requires dynamic R_X86_64_32 reloc which may overflow at runtime; recompile with -fPIC
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.

from intel linux.

Commit:
fix: harden libtagarray 0.0.7 update

- Build libtagarray static archives with PIC for Python extension linking
- Copy only active tagarray dimensions when returning record shapes
